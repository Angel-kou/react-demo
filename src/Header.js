import React from 'react'
import logo from './logo.svg';
import PropTypes from 'prop-types';



export class Header extends React.Component{
    shouldComponentUpdate(nextProps,nextState){
        return true;
    }

    componentWillReceiveProps(nextProps) {
        console.log('Component WILL RECEIVE PROPS!'+nextProps);
    }

    render(){
        return(
        <header className="App-header">
            <img ref={(el)=>{
                this.img = el;
            }}
                 src={logo} className="App-logo" alt="logo" />
            <h1 className="App-title">{this.props.appName}</h1>
        </header>
        )
    }
}

Header.propTypes = {
    appName: PropTypes.string
};

export default Header;