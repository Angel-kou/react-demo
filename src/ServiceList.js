import React, { Component } from 'react';
import request from './request'

class ServiceList extends Component {
        constructor(props) {
            super(props)
            this.state = {
                list: [],
                showList:true
            };
        }


        componentDidMount()
        {
            console.log(111)
            // fetch('https://api.github.com/search/repositories?q=topic:ruby+topic:rails')
            //     .then((response) => response.json())
                request().then((json)=>{
                    console.log(json)
                    this.setState({
                        list: json.items.slice(0, 10)
                    });
                });
        }

        render()
        {
            return (
                <div>
                    {
                        this.state.showList == true && this.state.list.map(item => {
                            return <li key={item.id}>{item.description}</li>
                        })
                    }
                </div>

            )
        }
}

export default ServiceList;