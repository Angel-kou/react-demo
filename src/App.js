import React, { Component } from 'react';
import './App.css';
import PropTypes from 'prop-types';
import Header from './Header.js';
import ServiceList from "./ServiceList";


class App extends Component {

    componentWillMount() {
        console.log('Component WILL MOUNT!')
    }
    componentDidMount() {
        console.log('Component DID MOUNT!')
    }
    componentWillReceiveProps(newProps) {
        console.log('Component WILL RECEIVE PROPS!')
    }
    shouldComponentUpdate(newProps, newState) {
        console.log('shouldComponentUpdate!')
        return true;
    }
    componentWillUpdate(nextProps, nextState) {
        console.log('Component WILL UPDATE!');
    }
    componentDidUpdate(prevProps, prevState) {
        console.log('Component DID UPDATE!')
    }
    componentWillUnmount() {
        console.log('Component WILL UNMOUNT!')
    }


    constructor(props){
        super(props);
        this.state={
            desc:"react demo",
            appName:"react demo 2018"
        };
        // this.onClick=this.changeAppDesc.bind(this);
    }

    changeAppDesc = () => {
        this.setState({
            desc:"react demo 2018"
        });
        // this.img.src = "/no-existing"
    }

    changeAppName = () => {
        this.setState({
            appName:"react redux"
        },()=>{
            console.log(this.state.appName);
        });

        // this.img.src = "/no-existing"
    }


  render() {
    return (
      <div className="App">
          <Header appName={this.state.appName}/>

        <p className="App-intro">
            {this.state.desc}
        </p>
          <button id="btn1" onClick={this.changeAppName}>change appName</button>
          <button id="btn2" onClick={this.changeAppDesc}>change desc</button>
          <ServiceList/>
      </div>

    );
  }

}

App.propTypes = {
    hello: PropTypes.string.isRequired
};

export default App;
