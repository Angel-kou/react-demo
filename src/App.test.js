jest.mock('./request')

import Adapter from 'enzyme-adapter-react-16'
import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import renderer from 'react-test-renderer';
import {shallow, mount,configure} from 'enzyme';
import Header from './Header.js';
import ServiceList from './ServiceList.js';

configure({adapter : new Adapter()})


it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App hello="world 2019" />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders App correctly', () => {
    const tree = renderer.create(<App hello="world 2019" />).toJSON();
    expect(tree).toMatchSnapshot();
});

it('renders App correctly', () => {
    const show = shallow(<App hello="world 2019" />);
    expect(show.find('p').text()).toEqual("react demo");
    // show.find('button').last().simulate('click');
    show.find('#btn2').simulate('click');
    expect(show.find('p').text()).toEqual("react demo 2018");

});

it('renders App correctly', () => {

    const service = mount(<ServiceList/>)

    return new Promise(resolve => setTimeout(resolve, 0)).then(() => {
        console.log(service.text(), '------')
        expect(service.text()).toEqual('12341234')
    });

});

