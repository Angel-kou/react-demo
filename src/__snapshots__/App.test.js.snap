// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders App correctly 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <img
      alt="logo"
      className="App-logo"
      src="logo.svg"
    />
    <h1
      className="App-title"
    >
      react demo 2018
    </h1>
  </header>
  <p
    className="App-intro"
  >
    react demo
  </p>
  <button
    id="btn1"
    onClick={[Function]}
  >
    change appName
  </button>
  <button
    id="btn2"
    onClick={[Function]}
  >
    change desc
  </button>
  <div />
</div>
`;
